src/Ninja/G1CP/Content/Fixes/Gamesave/fix203_LogEntryGrahamMerchant.d

```diff
@@ -0,0 +1,149 @@
+/*
+ * #203 Graham not listed as merchant
+ */
+
+/*
+ * Since applying and reverting both depend on finding the necessary symbols this is a toggle function
+ */
+func int G1CP_203_LogEntryGrahamMerchant_Toggle(var int apply) {
+    // Define possibly missing symbols locally
+    const int LOG_MISSION = 0;
+    const int LOG_NOTE    = 1;
+
+    // Parameters of this fix
+    const int    topicSection  = LOG_NOTE;
+    const string topicSymbName = "GE_TraderOC";
+    const string infoSymbName  = "DIA_Graham_Hello";
+    const string funcSymbName  = "DIA_Graham_Hello_Info";
+    const string hookSymbName  = "G1CP_203_LogEntryGrahamMerchant_Intercept";
+
+    // Retrieve the topic and entry strings once and modify the info function
+    const int    infoId  = -2; // -1 is reserved for invalid symbols
+    const int    topicId = -1;
+    const int    entryId = -1;
+    const string topic   = "G1CP invalid topic name";
+    const string entry   = "G1CP invalid topic entry";
+    if (infoId == -2) {
+        infoId = MEM_GetSymbolIndex(infoSymbName);
+        topicId = MEM_GetSymbolIndex(topicSymbName);
+        var int funcId; funcId = MEM_GetSymbolIndex(funcSymbName);
+        var int b_logentry_id; b_logentry_id = MEM_GetSymbolIndex("B_LogEntry");
+
+        // Do this only once (this is never reverted, i.e. session fix)
+        if (infoId == -1) || (funcId == -1) || (b_logentry_id == -1) || (topicId == -1) {
+            return FALSE;
+        };
+
+        // Find all calls to 'B_LogEntry' within the dialog function
+        const int bytes[2] = {zPAR_TOK_CALL<<24, -1};
+        var zCPar_Symbol needleSymb; needleSymb = _^(MEM_GetSymbolByIndex(b_logentry_id));
+        bytes[1] = needleSymb.content;
+        var int matches; matches = G1CP_FindInFunc(funcId, _@(bytes)+3, 5);
+
+        /* We are looking for:
+            zPAR_TOK_PUSHVAR topic
+            zPAR_TOK_PUSHVAR xxxx
+            zPAR_TOK_CALL    B_LogEntry
+        */
+
+        // Narrow down the search to calls with the correct topic as first argument to find the pushed entry
+        repeat(i, MEM_ArraySize(matches)); var int i;
+            var int pos; pos = MEM_ArrayRead(matches, i);
+            if (MEM_ReadByte(pos-10) == zPAR_TOK_PUSHVAR) && (MEM_ReadInt(pos-9) == topicId)
+            && (MEM_ReadByte(pos-5)  == zPAR_TOK_PUSHVAR) {
+                entryId = MEM_ReadInt(pos-4);
+                break;
+            };
+        end;
+        MEM_ArrayFree(matches);
+
+        // Check if we have found a valid entry
+        if (entryId <= 0) || (entryId >= MEM_Parser.symtab_table_numInArray) {
+            return FALSE;
+        };
```

Comment:
This if-condition can also be updated later. I will be much easier to use `G1CP_IsStringI(entryId)`.